cpp/CMakeLists.txt

```diff
@@ -322,15 +322,31 @@ endif(PER_THREAD_DEFAULT_STREAM)
 
 if(BUILD_TESTS)
     include(CTest)
-    include(ConfigureGoogleTest)
+    find_package(GTest)
+    
+    if(GTEST_ROOT)
+        message(STATUS "Google C++ Testing Framework (Google Test) found in ${GTEST_ROOT}")
+        add_subdirectory(${CMAKE_SOURCE_DIR}/tests)
+    else()
+        message(AUTHOR_WARNING "Google C++ Testing Framework (Google Test) not found, it will be built from source.")
+        include(ConfigureGoogleBenchmark)
+        add_subdirectory(${CMAKE_SOURCE_DIR}/tests)
```

Comment:
I think we need to run this regardless of whether we find or build gtest

Reply:
The include? I disagree. I can break it up into conditionals but it does all the checkout/build/install at include. I could split the checkout/build/install into a macro like the Blazing example. I'm not sure why specifically we'd need to do that though. I don't see anything in the module that isn't covered otherwise.

Reply:
Just the `add_subdirectory` command. That's what adds and configures the unit tests, so we presumably want to run that regardless of whether we find an existing package or build gbench ourselves.